### ISSUE TITLE
Renderdoc issues on Windows and OpenGL

### ISSUE THREAD

**Issue by DragosPopse** [libsdl-org/SDL#7263]
![image](https://user-images.githubusercontent.com/31741257/217430986-0c950edf-b0f3-4722-a6e4-d1bd70a9c629.png)

This is a weird issue, because I call SDL_GL_SetAttribute for the major, minor and profile mask and I set them to be 3, 3 and core. Also looking at the renderdoc error and the SDL source code, it seems that SDL itself doesn't use wglCreateContextAttribsARB that renderdoc expects.

**Comment by DragosPopse** [comment]
Coming back from more testing, someone had the same issue on Linux related to the GL context creation. It seems that the SDL_GL_SetAttributes calls get ignored. For windows, I got it working by creating a new opengl context and deleting the sdl created one. I'm using odin, so this is the solution so far 
```odin
dummycontext := sdl.GL_CreateContext(sdl_handle)
gl33_attribs := [?]i32 {
    win32.WGL_CONTEXT_MAJOR_VERSION_ARB, 3,
    win32.WGL_CONTEXT_MINOR_VERSION_ARB, 3,
    win32.WGL_CONTEXT_PROFILE_MASK_ARB, win32.WGL_CONTEXT_CORE_PROFILE_BIT_ARB,
    0,
}
syswm: sdl.SysWMinfo
sdl.GetVersion(&syswm.version)
sdl.GetWindowWMInfo(sdl_handle, &syswm)
win32.gl_set_proc_address(&win32.wglCreateContextAttribsARB, "wglCreateContextAttribsARB")
glcontext := win32.wglCreateContextAttribsARB(cast(win32.HDC)syswm.info.win.hdc, nil, &gl33_attribs[0])
sdl.GL_MakeCurrent(sdl_handle, cast(sdl.GLContext)glcontext)
sdl.GL_DeleteContext(dummycontext)
```
Using this, the `glGetString(GL_VERSION)` now returns the version of opengl that is setup in the context, as opposed to the latest available version the driver supports.

Overall it should be an easy fix on windows side, and probably on linux too. Let me know if this is an actual bug, and I can submit a PR

**Comment by slime73** [comment]
What's your window and context creation code which reproduces the issue for you?

Also keep in mind SDL_GL_SetAttribute needs to be called before window creation, on many platforms.

**Comment by DragosPopse** [comment]
![image](https://user-images.githubusercontent.com/31741257/217928704-c0dd872b-ac49-482c-bbe8-d347193e4ae5.png)
this was the code in question. I'm setting up the SDL_GL_SetAttribute stuff before the window. Sorry for the late response

**Comment by slime73** [comment]
I tested my own app just now with both SDL 2.24.0 and 2.26.3 and it's able to create Core Profile OpenGL contexts just fine (and works with RenderDoc). Are you sure that `when` condition in your code is being hit?

**Comment by DragosPopse** [comment]
yeah, i ran it through a debugger, but I'll test it without the condition too

**Comment by slouken** [comment]
We are scoping work for the SDL 3.2.0 release, so please let us know if this is a showstopper for you.

**Comment by DragosPopse** [comment]
Hi, no worries, this was an easy to work around bug, as just creating the opengl context by hand made it work. As not a lot of people encountered this, there is an off-chance that this could be a bindings issue, which is not a problem of sdl, so feel free to close this if it helps.

**Comment by slouken** [comment]
Okay, thanks for the update!

**Comment by jargv** [comment]
In case it's helpful to anyone, I was looking for a fix for this exact issue in a C++ project. After fiddling for a bit, adding the following line fixed the issue: 
```c++
  SDL_Init(SDL_INIT_VIDEO);
```
The project works great on Linux and Windows so I'm not even sure why that line is needed, or why things worked without it, but adding it fixed things up and RenderDoc works beautifully for me now, so :shrug:
